### PR TITLE
Add openLatch.reset to some tests

### PR DESCRIPTION
**Pull request by ortibazar**

### CODE REVIEW

**Comment by ktoso** [comment]
Refs https://github.com/akka/akka/issues/21719

(Please link back to issues a PR aims to solve :))

**Comment by ktoso** [comment]
OK TO TEST

**Comment by ktoso** [comment]
LGTM AFAICS, Thanks!

**Review by ktoso — APPROVED** [review]

**Comment by drewhk** [comment]
Can you explain why it needs to be reset?

**Comment by ortigali** [comment]
Issue: #21719 failure can be simulated changing test like below
```scala
    "increase the reset timeout after it transits to open again" in {
      val breaker = CircuitBreakerSpec.nonOneFactorCb()
      breaker().withCircuitBreaker(Future(throwException))
      checkLatch(breaker.openLatch)

      val e1 = intercept[CircuitBreakerOpenException] { breaker().withSyncCircuitBreaker(sayHi) }
      val shortRemainingDuration = e1.remainingDuration

      Thread.sleep(1000.millis.dilated.toMillis)
      checkLatch(breaker.halfOpenLatch)

      // transit to open again
      //  breaker().withCircuitBreaker(Future(throwException))
      breaker().withCircuitBreaker(Future{Thread.sleep(100);throwException})
      // checkLatch returns immediately without waiting for tripBreaker
      // because breaker.openLatch is already opened
      checkLatch(breaker.openLatch)

      val e2 = intercept[CircuitBreakerOpenException] { breaker().withSyncCircuitBreaker(sayHi) }
      val longRemainingDuration = e2.remainingDuration

      // breaker state is HalfOpen so longRemainingDuration is 0
      (shortRemainingDuration < longRemainingDuration) should ===(true)

    }
```

File: akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerSpec.scala [hunk 1]
```diff
@@ -107,6 +107,7 @@ class CircuitBreakerSpec extends AkkaSpec with BeforeAndAfter {
       val breaker = CircuitBreakerSpec.shortResetTimeoutCb()
       intercept[TestException] { breaker().withSyncCircuitBreaker(throwException) }
       checkLatch(breaker.halfOpenLatch)
+      breaker.openLatch.reset
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerSpec.scala]: Good.
The exception above will put it in Open followed by HalfOpen, and below we want to check that it transitions to Open again

File: akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerSpec.scala [hunk 2]
```diff
@@ -225,6 +227,7 @@ class CircuitBreakerSpec extends AkkaSpec with BeforeAndAfter {
       checkLatch(breaker.halfOpenLatch)
 
       // transit to open again
+      breaker.openLatch.reset
```
Comment (patriknw) [inline akka-actor-tests/src/test/scala/akka/pattern/CircuitBreakerSpec.scala]: The failing test was on the assert below:

```scala
 (shortRemainingDuration < longRemainingDuration) should ===(true)
```

That is testing the exponential backoff and the explanation would be that we checked too early, it had not performed the exponential increase yet.

**Review by patriknw — APPROVED** [review]
LGTM

**Comment by patriknw** [comment]
Thanks @ortigali !

**Comment by patriknw** [comment]
Refs #21719

**Pull request merged**